### PR TITLE
Add FlatContainer definition to SSZ

### PR DESCRIPTION
**Pull request by vbuterin**

### CODE REVIEW

File: specs/simple-serialize.md [hunk 1]
```diff
@@ -214,6 +214,25 @@ We now define Merkleization `hash_tree_root(value)` of an object `value` recursi
...
+* `pad_to_power_of_2(bitfield_bytes(value))` if `value` is a bitvector.
+* If the object is a vector of composite objects or a FlatContainer, use the algorithm below, where `elements` is the list of elements if the object is a vector, and an ordered list of its field values if it is a FlatContainer:
+
+```python
+output = b''
+for element in elements:
+    child = flatten(element)
+    # Pad output to a multiple of the length of the child, then add the child
+    output += b'\x00' * (-len(output) % len(child)) + child
+return output
```
Comment (vbuterin) [inline specs/simple-serialize.md]:
```suggestion
return pad_to_power_of_2(output)
```

File: specs/simple-serialize.md [hunk 2]
```diff
@@ -214,6 +214,25 @@ We now define Merkleization `hash_tree_root(value)` of an object `value` recursi
...
+* `pad_to_power_of_2(pack(value))` is `value` is a basic object or a vector of basic objects.
+* `pad_to_power_of_2(bitfield_bytes(value))` if `value` is a bitvector.
+* If the object is a vector of composite objects or a FlatContainer, use the algorithm below, where `elements` is the list of elements if the object is a vector, and an ordered list of its field values if it is a FlatContainer:
+
+```python
+output = b''
+for element in elements:
+    child = flatten(element)
+    # Pad output to a multiple of the length of the child, then add the child
+    output += b'\x00' * (-len(output) % len(child)) + child
```
Comment (djrtwo) [inline specs/simple-serialize.md]: Let's avoid negative number modulo arithmetic. This isn't well defined across all languages and we have removed such math from the spec for this reason in the past

Comment (vbuterin) [inline specs/simple-serialize.md]:
```suggestion
    if len(output) % len(child) != 0:
        output += b'\x00' * (len(child) - len(output) % len(child))
    output += child
```

Comment (vbuterin) [reply specs/simple-serialize.md]: @djrtwo does this look good?

Comment (djrtwo) [reply specs/simple-serialize.md]: yes

**Pull request closed without merging**